doc/content/development/contributing.md

```diff
@@ -0,0 +1,127 @@
+# Contributing
+
+The contributing guidelines here are mostly adapted from
+[MOOSE contributing guidelines](https://mooseframework.inl.gov/framework/contributing.html) and
+[Commit Often, Perfect Later, Publish Once: Git Best Practices](https://sethrobertson.github.io/GitBestPractices/)
+by Seth Robertson (CC BY-SA 3.0).
+
+## Code Standards
+
+Like MOOSE, Moltres uses the ClangFormat for formatting all C++ code in the repository. If you
+ever want to contribute changes to the Moltres repository, make you run the following command
+to automatically format any new code in your current branch relative to the supplied `<branch>`
+(defaults to HEAD if omitted).
+
+```bash
+git clang-format [<branch>]
+```
+
+The configuration file for the formatting is provided in the `.clang-format` file in the base
+directory. Refer to MOOSE [Code Standards](https://mooseframework.inl.gov/sqa/framework_scs.html)
+for more details on the code standard.
+
+## Referencing Issues
+
+In every pull request, we strongly recommend referencing an issue. This practice encourages robust
+discussions on whether the change is necessary and appropriate to Moltres. It also provides
+relevant context to the new changes proposed for quicker code review.
+
+## Work in a Fork
+
+Create your own [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) where you can
+make and commit your set of changes.
+
+### 1. Fork Moltres
+
+- Navigate to [https://github.com/arfc/moltres](https://github.com/arfc/moltres)
+- Click the "Fork" button near the upper right corner of the page.
+- Clone your new fork to your local machine using the following command (replace "username" with
+  your GitHub username):
+
+
+```bash
+git clone git@github.com:username/moltres.git
+```
+
+### 2. Add the `upstream` Remote
+
+Add the original Moltres repository as a remote named `upstream`:
+
+```bash
+cd moltres
+git remote add upstream git@github.com:arfc/moltres.git
+```
+
+### 3. Make Changes to Moltres
+
+Create a branch for your work:
+
+```bash
+git checkout -b branch_name
+```
+
+Make your changes and commit them to the branch you just created:
+
+```bash
+git add your_new_file.C your_new_file.h
+git commit -m "Your commit message here"
+```
+
+Ensure that your commit messages are insightful and descriptive such that they let people quickly
+understand your changes without having to read code. *Commit early and often* to create checkpoints
+that you can revert to if you break something while implementing your changes. Frequent commits
+also helps to keep your commits short and sweet.
+
+If you have been working on your changes for some time, there may have been new updates to Moltres
+implemented by others since the last time you pulled from the `upstream` remote. Merge those
+updates into your current branch by running the following commands while your current branch is
+checked out.
```

Comment:
I personally prefer to check out the main `devel` branch, and sync that before ever doing a merge. Because fetching and merging a remote can be tricky, this might be a lower barrier for entry. What do you think?

Reply:
To clarify, are you suggesting the following sequence?
```
git checkout devel
git fetch upstream
git merge upstream/devel
git checkout branch_name
git merge devel
```
I copied this set of instructions from MOOSE. I personally probably use both approaches without any preference nor issues because I'm familiar enough with the Git workflow. I'm not sure whether new Git users will struggle with the current instructions. The only issue I can foresee is if a merge conflict occurs, but I think new users will struggle with conflicts regardless of whether they're merging from upstream or local. Ultimately, I guess I don't have a preference.

Reply:
I like this sequence more, yes. I think this is a bit safer to do for new users, and it ensures that they have good `devel` branch hygiene. It's also more helpful if they have multiple feature branches that need to be updated.